backend/FwLite/MiniLcm/SyncHelpers/DiffCollection.cs

```diff
@@ -138,4 +221,120 @@ public static async Task<int> Diff<T>(
             },
             async (api, beforeEntry, afterEntry) => await replace(api, beforeEntry, afterEntry));
     }
+
+    private static BetweenPosition GetSurroundingStableIds<T>(int i, IList<T> current, IReadOnlyList<Guid> stable) where T : IOrderable
+    {
+        T? beforeEntity = default;
+        T? afterEntity = default;
+        for (var j = i - 1; j >= 0; j--)
+        {
+            if (stable.Contains(current[j].Id))
+            {
+                beforeEntity = current[j];
+                break;
+            }
+        }
+        for (var j = i + 1; j < current.Count; j++)
+        {
+            if (stable.Contains(current[j].Id))
+            {
+                afterEntity = current[j];
+                break;
+            }
+        }
+        return new BetweenPosition
+        {
+            Before = beforeEntity?.Id,
+            After = afterEntity?.Id
+        };
+    }
+
+    private static IEnumerable<PositionDiff> DiffPositions<T, TId>(
+        IList<T> before,
+        IList<T> after,
+        Func<T, TId> identity)
+    {
+        var beforeJson = new JsonArray(before.Select(item => JsonValue.Create(identity(item))).ToArray());
+        var afterJson = new JsonArray(after.Select(item => JsonValue.Create(identity(item))).ToArray());
+
+        if (JsonDiffPatcher.Diff(beforeJson, afterJson) is not JsonObject result)
```

Comment:
Essentially, all we need this library for is calculating a decent set of move operations.
Obviously, it calculates adds and deletes as well, but those are fairly trivial. So, if we wanted to we could refactor stuff to explicitly only use the moves this diff gives us. 🤷